client/src/includes/panels.ts

```diff
@@ -69,13 +73,67 @@ function initCollapsiblePanel(toggle: HTMLButtonElement) {
   content.addEventListener('beforematch', togglePanel.bind(null, true));
 }
 
+/**
+ * Initialises event handlers for collapsing / expanding all panels
+ */
+const initAllCollapsiblePanels = (toggles) => {
+  let collapseAllToggle = document.querySelector<HTMLButtonElement>(
+    '[data-all-panels-toggle]',
+  );
+
+  if (!collapseAllToggle) {
+    return;
+  }
+
+  let isExpanding = !collapseAllToggle.getAttribute('aria-expanded');
+
+  const togglesArray: ToggleAndContent[] = [];
+
+  toggles.forEach((toggle: HTMLButtonElement) => {
+    const content = document.querySelector<HTMLDivElement>(
+      `#${toggle.getAttribute('aria-controls')}`,
+    );
+    if (content) {
+      togglesArray.push({ toggle, content });
+    }
+  });
+
+  collapseAllToggle.addEventListener('click', () => {
+    if (!collapseAllToggle) {
+      return;
+    }
+
+    togglesArray.forEach((toggleObject) => {
+      if (isExpanding) {
+        toggleCollapsiblePanel(toggleObject.toggle, toggleObject.content, true);
+      } else {
+        toggleCollapsiblePanel(
+          toggleObject.toggle,
+          toggleObject.content,
+          false,
+        );
+      }
+    });
+    if (isExpanding) {
+      collapseAllToggle.setAttribute('aria-expanded', 'true');
+      isExpanding = false;
+      collapseAllToggle.innerText = 'Expand all sections';
+    } else {
+      collapseAllToggle.setAttribute('aria-expanded', 'false');
+      isExpanding = true;
+      collapseAllToggle.innerText = 'Collapse all sections';
```

Comment:
It might be better to put the text on the button as a data attribute -so it's easier to translate in the Django template.

Reply:
Presumably as well as setting the `innerText`? So something like `<button data-text="Collapse all sections">Collapse all sections</button>`

Reply:
Have now ensured all the text is translatable